athena-elasticsearch/README.md

```diff
@@ -0,0 +1,186 @@
+AmazonAthenaPreviewFunctionality and run any queries attempting to federate to this 
+connector, use a UDF, or SageMaker inference from that workgroup.
+
+## Nomenclature
+
+This document includes descriptions and explanations using Elasticsearch concepts and
+terminology:
+
+* **Domain** - A name Amazon Athena uses to associate with the endpoint of your Elasticsearch
+ instance and is also used as the database name. For Elasticsearch instances 
+ defined within the Amazon Elasticsearch Service, the domain is auto-discoverable. For all
+ other instances, a mapping between the domain name and endpoint will need to be provided.
+ 
+* **Index** - A database table defined in your Elasticsearch instance.
+
+* **Mapping** - If an index is a database table, then a mapping is its schema (i.e. definitions 
+of fields/attributes). This connector supports metadata retrieval directly from the 
+Elasticsearch instance, as well as from the Glue Data Catalog. If the connector finds a Glue 
+database and table matching your Elasticsearch domain and index names it will attempt to use it 
+for schema definition. We recommend creating your Glue table such that it is a superset of all 
+fields defined in your Elasticsearch index.
+
+* **Document** - A record within a database table.
+
+## Parameters
+
+The Amazon Athena Elasticsearch Connector exposes several configuration options via Lambda 
+environment variables:
+
+1. **disable_glue** - (Optional) If present, with any value, the connector will no longer 
+attempt to retrieve supplemental metadata from Glue.
+
+2. **auto_discover_endpoint** - true/false (true is the default value). For Amazon 
+Elasticsearch Service the domain-names and their associated endpoints are auto-discoverable.
+For any other service, the associated domain-endpoints must be specified in the **domain_mapping** 
+variable. This also determines which credentials will be used to access the endpoint. If 
+**auto_discover_endpoint**=**true**, then AWS credentials will be used. Otherwise, 
+username/password credentials retrieved from Amazon Secrets Manager via the **secret_name** 
+variable will be used.
+
+3. **domain_mapping** - Used only when **auto_discover_endpoint**=**false**, 
+this is the mapping between the domain names and their associated endpoints (in the format:
+`domain1=endpoint1,domain2=endpoint2`). 
+
+    ```                        
+        Example: 
+            
+        movies=search-movies-ne...qu.us-east-1.es.amazonaws.com
+    ```
+
+4. **secret_name** - Used only when **auto_discover_endpoint**=**false** to retrieve the 
+username/password credentials from Amazon Secrets Manager in order to connect to the
+Elasticsearch instance. The username and password can be stored in Secrets Manager  using the 
+following formats:
+    
+    * Two Key-Value Pairs:
+    `"username": <username>, "password": <password>`
+    * Single Key-Value: `<username>: <password>`
+    
```

Comment:
Let's discuss further. Unlike other connectors, this one does not use a connection string so it may make it counter intuitive and confusing to the users to have a connection string as part of the domain mapping. This is how the client for non-Amazon ES users is injected with username/password credentials:            
```java
final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(username, password));

clientBuilder.setHttpClientConfigCallback(httpClientBuilder -> httpClientBuilder
                    .setDefaultCredentialsProvider(credentialsProvider));
```

Reply:
will change to connection str